l1-contracts/test/foundry/l1/integration/L1GatewayTests.t.sol

```diff
@@ -244,102 +241,62 @@ contract GatewayTests is L1ContractDeployer, ZKChainDeployer, TokenDeployer, L2T
         vm.stopBroadcast();
     }
 
-    function test_registerAlreadyDeployedZKChain() public {
-        gatewayScript.governanceRegisterGateway();
-
-        IChainTypeManager stm = IChainTypeManager(l1Script.getCTM());
-        IBridgehub bridgehub = IBridgehub(l1Script.getBridgehubProxyAddress());
-        address owner = Ownable(address(bridgeHub)).owner();
-
-        {
-            uint256 chainId = currentZKChainId++;
-            bytes32 baseTokenAssetId = DataEncoding.encodeNTVAssetId(chainId, ETH_TOKEN_ADDRESS);
-
-            address chain = _deployZkChain(
-                chainId,
-                baseTokenAssetId,
-                address(bridgehub.sharedBridge()),
-                owner,
-                stm.protocolVersion(),
-                stm.storedBatchZero(),
-                address(bridgehub)
-            );
-
-            address stmAddr = IZKChain(chain).getChainTypeManager();
-
-            vm.startBroadcast(owner);
-            bridgeHub.addChainTypeManager(stmAddr);
-            bridgeHub.addTokenAssetId(baseTokenAssetId);
-            bridgeHub.registerAlreadyDeployedZKChain(chainId, chain);
-            vm.stopBroadcast();
-
-            address bridgeHubStmForChain = bridgeHub.chainTypeManager(chainId);
-            bytes32 bridgeHubBaseAssetIdForChain = bridgeHub.baseTokenAssetId(chainId);
-            address bridgeHubChainAddressdForChain = bridgeHub.getZKChain(chainId);
-            address bhAddr = IZKChain(chain).getBridgehub();
-
-            assertEq(bridgeHubStmForChain, stmAddr);
-            assertEq(bridgeHubBaseAssetIdForChain, baseTokenAssetId);
-            assertEq(bridgeHubChainAddressdForChain, chain);
-            assertEq(bhAddr, address(bridgeHub));
-        }
-
-        {
-            uint256 chainId = currentZKChainId++;
-            bytes32 baseTokenAssetId = DataEncoding.encodeNTVAssetId(chainId, ETH_TOKEN_ADDRESS);
-            address chain = _deployZkChain(
-                chainId,
-                baseTokenAssetId,
-                address(bridgehub.sharedBridge()),
-                owner,
-                stm.protocolVersion(),
-                stm.storedBatchZero(),
-                address(bridgehub.sharedBridge())
-            );
-
-            address stmAddr = IZKChain(chain).getChainTypeManager();
-
-            vm.startBroadcast(owner);
-            bridgeHub.addTokenAssetId(baseTokenAssetId);
-            vm.expectRevert(
-                abi.encodeWithSelector(IncorrectBridgeHubAddress.selector, address(bridgehub.sharedBridge()))
-            );
-            bridgeHub.registerAlreadyDeployedZKChain(chainId, chain);
-            vm.stopBroadcast();
-        }
+    function test_migrateBackChain() public {
```

Comment:
are you sure it should be L1GatewayTest? It is more like an L2 one

Reply:
migrateBack is GW->L1 migration, it checks the finalizeDeposit part on L1. It should be finishMigrateBackChain